networking_generic_switch/generic_switch_mech.py

```diff
@@ -163,34 +166,24 @@ def delete_network_postcommit(self, context):
 
         if provider_type == 'vlan' and segmentation_id:
             # Delete vlan on all switches from this driver
+            exc_info = None
             for switch_name, switch in self._get_devices_by_physnet(physnet):
                 try:
-                    # NOTE(mgoddard): The del_network method was modified to
-                    # accept the network ID. The switch object may still be
-                    # implementing the old interface, so retry on a TypeError.
-                    try:
-                        switch.del_network(segmentation_id, network['id'])
-                    except TypeError:
-                        if not self.warned_del_network:
-                            msg = (
-                                'The del_network device method should accept '
-                                'the network ID. Falling back to just the '
-                                'segmentation ID for %(device)s. This '
-                                'transitional support will be removed in the '
-                                'Rocky release')
-                            LOG.warn(msg, {'device': switch_name})
-                            self.warned_del_network = True
-                        switch.del_network(segmentation_id)
+                    switch.del_network(segmentation_id, network['id'])
                 except Exception as e:
                     LOG.error("Failed to delete network %(net_id)s "
                               "on device: %(switch)s, reason: %(exc)s",
                               {'net_id': network['id'],
                                'switch': switch_name,
                                'exc': e})
+                    # Save any exceptions for later reraise.
+                    exc_info = sys.exc_info()
```

Comment:
Doesn't this only save the last exception?

Reply:
Yes. If there is more than one that's ok, we just want to reraise something, but continue deleting.